app/src/main/java/de/westnordost/streetcomplete/quests/surface/DetailRoadSurfaceForm.kt

```diff
@@ -0,0 +1,100 @@
+package de.westnordost.streetcomplete.quests.surface
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import androidx.appcompat.app.AlertDialog
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.quests.AImageListQuestAnswerFragment
+import de.westnordost.streetcomplete.quests.OtherAnswer
+import de.westnordost.streetcomplete.util.TextChangedWatcher
+import de.westnordost.streetcomplete.view.Item
+
+class DetailRoadSurfaceForm  : AImageListQuestAnswerFragment<String, DetailSurfaceAnswer>() {
+
+    override val items: List<Item<String>>
+        get() = if (osmElement!!.tags["surface"] == "paved")
+            (PAVED_SURFACES + UNPAVED_SURFACES + GROUND_SURFACES).toItems()
+        else
+            (UNPAVED_SURFACES + GROUND_SURFACES + PAVED_SURFACES).toItems()
```

Comment:
Ah! Now I understand why there was this solution with topItems. If it was paved, you first wanted to hide the non-paved items and only on another button click, show them. Sorry for the previous remark then, then it was already fine.

But it is fine like this as well.

Reply:
I think I like unified list more, I plan to switch to it also be normal surface quest in my fork. Less clicking to answer the question.